src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+            if (fields !== this.state.fields && this.iAmMounted) {
+                this.setState({fields});
+            }
+        });
+    }
+
+    render() {
+        var {fields}= this.state;
+        return (
+            <div style={{padding: 10}}>
+                <FormPanel
+                    groupKey={gkey}
+                    onSubmit={(request) => onSearchSubmit(request)}
+                    onCancel={hideSearchPanel}>
+                    <CatalogSelectView fields={fields}/>
+                </FormPanel>
+            </div>
+        );
+    }
+}
+
+function onSearchSubmit(request) {
+    console.log('request ' + JSON.stringify(request));
+
+    if (request.Tabs === 'catalog') {
+        const wp = parseWorldPt(request[ServerParams.USER_TARGET_WORLD_PT]);
+        if (!wp && (request.spatial === SpatialMethod.Cone.value
+            || request.spatial === SpatialMethod.Box.value
+            || request.spatial === SpatialMethod.Elliptical.value)) {
+            showInfoPopup('Target is required');
+            return;
+        }
+        doCatalog(request);
+    }
+    else if (request.Tabs === 'loadcat') {
+        doLoadTable(request);
+    }
+    else if (request.Tabs === 'vosearch') {
+        doVoSearch(request);
+    }
+    else {
+        console.log('request no supported');
+    }
+}
+function hideSearchPanel() {
+    dispatchHideDropDown();
+}
+
+function doCatalog(request) {
+
+    const conesize = convertAngle('deg', 'arcsec', request.conesize);
+    var title = request.project + '-' + request.cattable + ' (' + request.spatial;
+    if (request.spatial === SpatialMethod.Box.value || request.spatial === SpatialMethod.Cone.value || request.spatial === SpatialMethod.Elliptical.value) {
+        title += ':' + conesize;
+    }
+    title += ')';
+    var tReq = TableRequest.newInstance({
+
+        id: 'GatorQuery',
+        title: title,
+        SearchMethod: request.spatial,
+        catalog: request.cattable,
+        RequestedDataSet: request.catalog,
+        use: 'catalog_overlay',
+        catalogProject: request.project
+
+    });
+
+    // change and merge others parameters in request if elliptical
+    // plus change spatial name to cone
+    // (Gator search method for elliptical is cone)
+    if (request.spatial === SpatialMethod.Elliptical.value) {
+
+        const pa = get(request, 'posangle', 0);
+        const ar = get(request, 'axialratio', 0.26);
+
+        // see PA and RATIO string values in edu.caltech.ipac.firefly.server.catquery.GatorQuery
+        merge(tReq, {'posang': pa, 'ratio': ar});
+    }
+
+    if (request.spatial === SpatialMethod.Cone.value
+        || request.spatial === SpatialMethod.Box.value
+        || request.spatial === SpatialMethod.Elliptical.value) {
+        merge(tReq, {[ServerParams.USER_TARGET_WORLD_PT]: request[ServerParams.USER_TARGET_WORLD_PT]});
+        if (request.spatial === SpatialMethod.Box.value) {
+            merge(tReq, {'size': conesize});
```

Comment:
yes, agree. I was so enthusiast to use merge function that forgot to 
keep it simpler ;-)

On 5/14/16 11:18 PM, Trey Roby wrote:
>
> In src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx 
> <https://github.com/Caltech-IPAC/firefly/pull/85#discussion_r63286841>:
>
> > +    // (Gator search method for elliptical is cone)
> > +    if (request.spatial === SpatialMethod.Elliptical.value) {
> > +
> > +        const pa = get(request, 'posangle', 0);
> > +        const ar = get(request, 'axialratio', 0.26);
> > +
> > +        // see PA and RATIO string values in edu.caltech.ipac.firefly.server.catquery.GatorQuery
> > +        merge(tReq, {'posang': pa, 'ratio': ar});
> > +    }
> > +
> > +    if (request.spatial === SpatialMethod.Cone.value
> > +        || request.spatial === SpatialMethod.Box.value
> > +        || request.spatial === SpatialMethod.Elliptical.value) {
> > +        merge(tReq, {[ServerParams.USER_TARGET_WORLD_PT]: request[ServerParams.USER_TARGET_WORLD_PT]});
> > +        if (request.spatial === SpatialMethod.Box.value) {
> > +            merge(tReq, {'size': conesize});
>
> merge or Object.assign is unnecessary for stuff like this. Just simply 
> do this:
>
> |tReq.size= conesize' |
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/Caltech-IPAC/firefly/pull/85/files/8bdfd737fd8631b45d10aff1ebb2742bee675e9e#r63286841>
>

-- 
Emmanuel Joliet
NASA/IPAC  Infrared Science Archive
MS 100-22, Caltech, Pasadena, CA 91125
Office: 170
Email:  ejoliet@ipac.caltech.edu
Phone:  (626) 395-1489